File: packages/EvoHtmlToPdf-v4.12/Samples/NET_4.0/CS/WinForms/EvoHtmlToPdfDemo/ConvertHtmlString.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;

using EvoPdf;


namespace EvoHtmlToPdfDemo
{
    public partial class ConvertHtmlString : Form
    {
        public ConvertHtmlString()
        {
            InitializeComponent();
        }

        private void btnConvert_Click(object sender, EventArgs e)
        {
            this.Cursor = Cursors.WaitCursor;

            try
            {
                PdfConverter pdfConverter = new PdfConverter();

                // set the license key
                pdfConverter.LicenseKey = "B4mYiJubiJiInIaYiJuZhpmahpGRkZE=";

                // save the PDF bytes in a file on disk
                string outFilePath = Path.Combine(Application.StartupPath, "ConvertHtmlString.pdf");

                pdfConverter.SavePdfFromHtmlStringToFile(textBoxHtmlString.Text, outFilePath, textBoxBaseUrl.Text.Trim());
                
                DialogResult dr = MessageBox.Show("Open the rendered file in an external viewer?", "Open Rendered File", MessageBoxButtons.YesNo);
                if (dr == DialogResult.Yes)
                {
                    System.Diagnostics.Process.Start(outFilePath);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }
            finally
            {
                this.Cursor = Cursors.Arrow;
            }
        }
    }
}

File: packages/EvoHtmlToPdf-v4.12/Samples/NET_4.0/VB/WinForms/EvoHtmlToPdfDemo/ConvertHtmlString.vb
Imports System.ComponentModel
Imports System.Text
Imports System.IO

Imports EvoPdf


Namespace EvoHtmlToPdfDemo
	Partial Public Class ConvertHtmlString
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnConvert_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnConvertMerge.Click
			Me.Cursor = Cursors.WaitCursor

			Try
				Dim pdfConverter As New PdfConverter()

				' set the license key
                pdfConverter.LicenseKey = "B4mYiJubiJiInIaYiJuZhpmahpGRkZE="

				' save the PDF bytes in a file on disk
				Dim outFilePath As String = Path.Combine(Application.StartupPath, "ConvertHtmlString.pdf")

				pdfConverter.SavePdfFromHtmlStringToFile(textBoxHtmlString.Text, outFilePath, textBoxBaseUrl.Text.Trim())

				Dim dr As DialogResult = MessageBox.Show("Open the rendered file in an external viewer?", "Open Rendered File", MessageBoxButtons.YesNo)
				If dr = DialogResult.Yes Then
					Process.Start(outFilePath)
				End If
			Catch ex As Exception
				MessageBox.Show(ex.Message)
				Return
			Finally
				Me.Cursor = Cursors.Arrow
			End Try
		End Sub
	End Class
End Namespace